source/components/Autocomplete.js

```diff
@@ -192,24 +196,44 @@ class AutocompleteBase extends Component<AutocompleteProps, State> {
       !maxSelections ||
       (canMoreOptionsBeSelected && areFilteredOptionsAvailable)
     ) {
-      if (!selectedOption) return;
-      const option = selectedOption.trim();
-      const optionCanBeSelected =
-        (selectedOptions.indexOf(option) < 0 && !multipleSameSelections) ||
-        multipleSameSelections;
-
-      if (option && optionCanBeSelected && isOpen) {
-        const newSelectedOptions = _.concat(selectedOptions, option);
-        this.selectionChanged(newSelectedOptions, event);
-        this.setState({ selectedOptions: newSelectedOptions, isOpen: false });
+      if (!selectedOption || !selectedOption.length) return;
+      const option = _.isString(selectedOption) ? selectedOption.trim() : selectedOption;
+      let newSelectedOptions = [];
+      let skipValuesSelection = false;
+      if (option && Array.isArray(option)) {
+        newSelectedOptions = selectedOptions;
+        option.forEach(item => {
+          const optionCanBeSelected =
+            (selectedOptions.indexOf(item) < 0 &&
+              newSelectedOptions.indexOf(item) < 0 &&
+              filteredOptions.indexOf(item) &&
+              !multipleSameSelections) ||
+            multipleSameSelections;
+          if (filteredOptions.indexOf(item) < 0) {
+            skipValuesSelection = true;
+            return;
+          }
+          if (item && optionCanBeSelected && isOpen && !skipValuesSelection) {
+            newSelectedOptions = _.concat(newSelectedOptions, item);
+          }
+        });
+      } else {
+        const optionCanBeSelected =
+          (selectedOptions.indexOf(option) < 0 && !multipleSameSelections) ||
+          multipleSameSelections;
+        if (option && optionCanBeSelected && isOpen) {
+          newSelectedOptions = _.concat(selectedOptions, option);
```

Comment:
spreading is probably easier and you could remove the `concat` import: `[...selectedOptions, option]`

Reply:
Updated

Reply:
> Hey @aleksandardjordjeviciohk, great work – the pasting works!
> I added a few ideas how to make the logic more readable – also a few code comments could help here :)
> 
> One UX thing I noted: wasn't the idea that when you paste a series of words and one in the middle is incorrect, that this word is the last visible + the options popup shows "no results"? – at the moment only the correct words up until the first incorrect are pasted without any warning etc.

I have added a small fix which includes also adding first wrong word into field and showing popup message 'no results' with a cursor focus on that last word.